src/Experimental/Template.fs

```diff
@@ -67,42 +67,42 @@ let extract_template (nsx : string) (bindings : Map<string,Binder>) (xml : Xml)
         loop tail (Xml(acc @ (subst)))
 
     loop nodelist (Xml([]))
-  extract_template' (xml) (Xml([]))
+  extractTemplateAux (xml) (Xml([]))
 
 /// Bind the matching elements, given the namespace 'ns', a XML/HTML document 'node'
 /// and a 'bindings' map.
 let bind (ns : string, node : Xml, bindings : Map<string,Binder>) =
-  extract_template ns bindings node
+  extractTemplate ns bindings node
 
 /// An attribute array; an array of key-value pairs
 type Attributes = (string*string)[]
 
 /// Write the start of the tag 's' to the string writer
-let write_begin_tag (s : string) n (a : Attributes) (stream : StringWriter) =
+let writeBeginTag (s : string) n (a : Attributes) (stream : StringWriter) =
   let tt = Array.fold( fun z (x,y)  -> z + (sprintf "%s=\"%s\" " x y) ) " " a
   if tt = " " then stream.Write("<{0}>", s)
   else stream.Write("<{0} {1}>",s,tt)
 
 /// Write the end of tag 's' to the string writer
-let write_close_tag (s : string) n a (stream : StringWriter) =
+let writeCloseTag (s : string) n a (stream : StringWriter) =
   stream.Write("</{0}>", s)
 
 /// Recursively convert the Xml structure to text into the string writer
-let rec xml_to_string1 (Xml(nodes)) (stream : StringWriter) =
+let rec xmlToString1 (Xml(nodes)) (stream : StringWriter) =
```

Comment:
Isn't it obvious from the types that the function takes that it's to a string writer? Anyway, I'd welcome @ademar 's opinion here, with regards to making this method private, so that it can have a 'simple' name -- writing the XML surely is something the user wouldn't do?